### PR TITLE
Trigger compaction after object pruning

### PR DESCRIPTION
**Pull request by sadhansood**

### CODE REVIEW

**Comment by lxfind** [comment]
Have we confirmed whether object pruning works with lamport versioning? i.e. latest 2 versions instead of latest version - 2? We turned off pruning because of concern on that.

**Comment by sadhansood** [comment]
> Have we confirmed whether object pruning works with lamport versioning? i.e. latest 2 versions instead of latest version - 2? We turned off pruning because of concern on that.

Yes most definitely. We never look at the value of the object version itself, but just the fact that we don't touch the latest 2 versions of an object (in the order they appear in the table during scan) and delete everything else.

File: crates/sui-core/src/authority/authority_store_pruner.rs [hunk 1]
```diff
@@ -139,9 +139,15 @@ impl AuthorityStorePruner {
                 tokio::select! {
                     _ = prune_interval.tick() => {
                         info!("Starting pruning of objects table");
-                        let perpetual_db = perpetual_db.clone();
-                        let num_pruned = Self::prune_objects(num_versions_to_retain, perpetual_db);
+                        let num_pruned = Self::prune_objects(num_versions_to_retain, perpetual_db.clone());
                         info!("Finished pruning with total object versions pruned = {}", num_pruned);
+                        let min_key = ObjectKey(ObjectID::ZERO, SequenceNumber::MIN);
+                        let max_key = ObjectKey(ObjectID::MAX, SequenceNumber::MAX);
+                        if let Ok(()) = perpetual_db.objects.compact_range(&min_key, &max_key) {
```
Comment (longbowlu) [inline crates/sui-core/src/authority/authority_store_pruner.rs:146]: can we record how long this takes in log and/or metrics?

**Review by longbowlu — APPROVED** [review]

File: crates/sui-core/src/authority/authority_store_pruner.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-core/src/authority/authority_store_pruner.rs:146]: Sure let me send a trivial follow up PR. Thanks

Comment (sadhansood) [reply crates/sui-core/src/authority/authority_store_pruner.rs:147]: Added duration as suggested

**Pull request merged**